Control/DeepSeq.hs

```diff
@@ -5,16 +5,24 @@
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE TypeOperators #-}
 # if MIN_VERSION_array(0,4,0)
+#if __GLASGOW_HASKELL__ >= 708
+-- We can't use Safe because Data.Coerce is not marked safe.
+{-# LANGUAGE Trustworthy #-}
+#else
 {-# LANGUAGE Safe #-}
+#endif
 # endif
 #endif
 #if __GLASGOW_HASKELL__ >= 706
 {-# LANGUAGE PolyKinds #-}
 #endif
+#if __GLASGOW_HASKELL__ >= 708
+{-# LANGUAGE ScopedTypeVariables #-}
+#endif
```

Comment:
I'd prefer to enable `ScopedTypeVariables` unconditionally (without an `#ifdef`), since all versions of GHC that we test against support `ScopedTypeVariables` anyway.